frontend/src/views/Game.tsx

```diff
@@ -38,24 +42,37 @@ function GamePage() {
 
   const firstProblem = problems?.[0];
 
+  const onSecondaryPanelSizeChange = () => {
+    const event = new Event('secondaryPanelSizeChange');
+    window.dispatchEvent(event);
```

Comment:
I'd agree that it'd be nice to be able to avoid using events for example with calling `editor.layout` directly here. However, I'm not sure whether that's actually possible or not - maybe one option is pass in a custom callback to editorDidMount that somehow is triggered whenever the splitter resize is triggered? The difficulty though is that you need a reference to `editor` in order to call `layout`, and it seems like that might only be made available once on editorDidMount, hence the need for events. The best solution would obviously be like an `automaticLayout = "true"` flag, but I guess that doesn't exist with the monaco editor we're using haha

Reply:
I haven't figured out a solution that doesn't use events yet, so I can't be sure whether it's possible or not. However, the vast majority of online solutions for adjustable Monaco editors recommend using an event listener and calling `editor.layout()` (other than just adding the `automaticLayout` property but as Alan said that doesn't work for our editor).